File: NominasSpringBootFinal/src/main/java/com/nomina/nomina/service/EmpleadoService.java
package com.nomina.nomina.service;




import com.nomina.nomina.entity.Empleado;
import com.nomina.nomina.entity.Nomina;

import java.util.List;

public interface EmpleadoService {

    Empleado traerEmpleado(int id);
    Nomina traerSueldo(String dni);
    List<Empleado> listarEmpleados();
    void guardarEmpleado(Empleado empleado);
    void eliminarEmpleado(int id);
}

File: Nominas/src/app/services/empleado.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Empleado } from '../interfaces/empleado';

@Injectable({
  providedIn: 'root'
})

export class EmpleadoService {

  private baseUrl = 'http://localhost:8082/controlador';

  constructor(private http:HttpClient) { }

  listarEmpleados(){
    return new Promise<Empleado[]>(resolve=>{
      this.http.get<Empleado[]>(`${this.baseUrl}`+`/lista`).subscribe(resp=>{
        resolve(resp);
      });
    });
  }

  

  obtenerSueldo(dni:String): Observable<any> {
    return this.http.post(`${this.baseUrl}`+`/sueldo`,dni);
  }

  guardarEmpleado(empleado: Empleado): Observable<any> {
    return this.http.post(`${this.baseUrl}`+'/guardarEmpleado', empleado);
  }

  borrarEmpleado(id: number): Observable<any> {
    return this.http.delete(`${this.baseUrl}/eliminar/${id}`, { responseType: 'text' });
  }

  cogerEmpleado(id: number): Observable<any> {
    return this.http.get(`${this.baseUrl}/traerEmpleado/${id}`);
  }
  
}  